sw/device/silicon_creator/lib/otbn_util.c

```diff
@@ -0,0 +1,124 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/otbn_util.h"
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/silicon_creator/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/otbn.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+void otbn_init(otbn_t *ctx) {
+  *ctx = (otbn_t){
+      .app = {0},
+      .app_is_loaded = false,
+      .error_bits = kOtbnErrBitsNoError,
+  };
+}
+
+uint32_t otbn_get_internal_err_bits(otbn_t *ctx) { return ctx->error_bits; }
+
+rom_error_t otbn_func_ptr_to_imem_addr(const otbn_t *ctx, otbn_ptr_t ptr,
+                                       uint32_t *imem_addr_otbn) {
+  const uint32_t *app_imem_start_addr = ctx->app.imem_start;
+  const uint32_t *app_imem_end_addr = ctx->app.imem_end;
+
```

Comment:
can be inlined (similar to `otbn_data_ptr_to_dmem_addr`), right?

Reply:
This function can go away, it's not needed any more without the START_ADDR register.